Lesson 2 Пятнашки/15.js

```diff
@@ -1,35 +1,84 @@
+/*
+	The same code as Kyrylo Yaremchuk provided
+	Don't know who is real author
+*/
+
+/*
+  Don't use such school pascal lessons style
+  It is very hard to get what is going on in your code :)
+
+  A lot of hardcoded values
+
+  Why arr is global if you use it only in the one routine
+*/
 let arr = [], box, ei,ej;
-function swap(arr,i1,j1,i2,j2){				
+
+function swap(arr,i1,j1,i2,j2){		
+	/*
+    	without keyword it is a global variable
+  	*/		
 	t = arr[i1][j1];
 	arr[i1][j1] = arr[i2][j2];
 	arr[i2][j2] = t;
 }
+
+/*
+  Use of DOM Level 2 events (addEventListener) is better practice
+*/
 window.onload = function() {				
 	box = document.getElementById("box");
 	newGame();				
 	document.getElementById("reset").onclick = newGame;						
 }
+
+/*
+  Code is very messy
+  You have to divide logic for more routines
+  instead of having all in one
+*/
 function cellClick(event) {
+	/*
+	    No need for redeclare of event
+	    It is old style when you couldn't be sure that event is passed
+	    As far as you handle event on the right element you get right vent object
+
+	    Bad idea to use element`s id as data storage
+	    event.target instead of event.srcElement or just 'this' as long as you hadn't overrided context for your handler
+ 	*/
 	var event = event || window.event,
 		el = event.srcElement || event.target,
 		i = el.id.charAt(0),
 		j = el.id.charAt(2);
+
+	/*
+		A lot of nested constuctions
+	*/
 	if((i == ei && Math.abs(j - ej) == 1) || (j == ej && Math.abs(i - ei) == 1)){					
 		document.getElementById(ei + " " + ej).innerHTML = el.innerHTML;
 		el.innerHTML = "";
 		ei = i;
 		ej = j;
 		let q = true;
+
+		/*
+	      Very strange win verification
+	      Could use much more simplier
+	    */
 		for(i = 0; i < 4; ++i)
 			for(j = 0; j < 4; ++j)
 				if(i + j != 6 && document.getElementById(i + " " + j).innerHTML != i*4 + j + 1){
 					q = false;
 					break;
```

Comment:
This break will return only from 1 loop
But once you got your flag false you have to break both